bootstrap/internal/controller/agent_controller.go

```diff
@@ -137,23 +148,73 @@ func (r *AgentReconciler) getClusterName(ctx context.Context, agent *aiv1beta1.A
 	return clusterName, nil
 }
 
-func getIngitionConfigOverride() string {
-	ignition := `{
-				"ignition": { "version": "3.1.0" },
-				"storage": {
-                  "files": [
-					  {
-		                "path": "/run/cluster-api/bootstrap-success.complete",
-				        "mode": 420,
-				        "contents": {
-							"source": "data:text/plain;charset=utf-8;base64,c3VjY2Vzcw=="
-						}
-				      }
-			      ]
-				}
+func (r *AgentReconciler) getIgnitionConfigOverride(ctx context.Context, bootstrapConfig *bootstrapv1alpha1.AgentBootstrapConfig) (string, error) {
```

Comment:
In case we go with ignition, can we extract ignition related stuff to its package? maybe `bootstrap/internal/ignition`

Could you also please do add basic tests?